### PR TITLE
add tests for time_on_page edge cases

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR adds tests for which https://github.com/plausible/analytics/pull/3446 fails unless https://github.com/ruslandoga/analytics/pull/183 is merged.

### Tests
- [x] Automated tests have been added

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: test/plausible_web/controllers/api/stats_controller/pages_test.exs [hunk 1]
```diff
@@ -772,6 +772,35 @@ defmodule PlausibleWeb.Api.StatsController.PagesTest do
              ]
     end
 
+    test "doesn't calculate time on page with only single page visits", %{conn: conn, site: site} do
+      populate_stats(site, [
+        build(:pageview, pathname: "/", user_id: @user_id, timestamp: ~N[2021-01-01 00:00:00]),
+        build(:pageview, pathname: "/", user_id: @user_id, timestamp: ~N[2021-01-01 00:10:00])
+      ])
+
+      assert [%{"name" => "/", "time_on_page" => nil}] =
```
Comment (ruslandoga) [inline test/plausible_web/controllers/api/stats_controller/pages_test.exs:781]: `NULL` happens due to division by zero here https://github.com/plausible/analytics/blob/f977351ae2ba4d79f432978317a49e64dcaafa89/lib/plausible/stats/breakdown.ex#L235

File: test/plausible_web/controllers/api/stats_controller/pages_test.exs [hunk 2]
```diff
@@ -772,6 +772,35 @@ defmodule PlausibleWeb.Api.StatsController.PagesTest do
...
+    test "ignores page refresh when calculating time on page", %{conn: conn, site: site} do
+      populate_stats(site, [
+        build(:pageview, user_id: @user_id, timestamp: ~N[2021-01-01 00:00:00], pathname: "/"),
+        build(:pageview, user_id: @user_id, timestamp: ~N[2021-01-01 00:01:00], pathname: "/"),
+        build(:pageview, user_id: @user_id, timestamp: ~N[2021-01-01 00:02:00], pathname: "/"),
+        build(:pageview, user_id: @user_id, timestamp: ~N[2021-01-01 00:03:00], pathname: "/exit")
+      ])
+
+      assert [
+               %{"name" => "/", "time_on_page" => _three_minutes = 180.0},
```
Comment (ruslandoga) [inline test/plausible_web/controllers/api/stats_controller/pages_test.exs:796]: Current approach divides the session duration over page transition count, and in this tests there's only one page transition (/ -> /exit).

File: test/plausible_web/controllers/api/stats_controller/top_stats_test.exs [hunk 3]
```diff
@@ -296,6 +296,42 @@ defmodule PlausibleWeb.Api.StatsController.TopStatsTest do
...
+    test "doesn't calculate time on page with only single page visits", %{conn: conn, site: site} do
+      populate_stats(site, [
+        build(:pageview, pathname: "/", user_id: @user_id, timestamp: ~N[2021-01-01 00:00:00]),
+        build(:pageview, pathname: "/", user_id: @user_id, timestamp: ~N[2021-01-01 00:10:00])
+      ])
+
+      filters = Jason.encode!(%{page: "/"})
+      path = "/api/stats/#{site.domain}/top-stats?period=day&date=2021-01-01&filters=#{filters}"
+
+      assert %{"name" => "Time on page", "value" => 0} ==
```
Comment (ruslandoga) [inline test/plausible_web/controllers/api/stats_controller/top_stats_test.exs:309]: Aggregate is similar, except here `Inf` gets turned into `NULL` before being returned https://github.com/plausible/analytics/blob/f977351ae2ba4d79f432978317a49e64dcaafa89/lib/plausible/stats/aggregate.ex#L113

**Review by ukutaht — APPROVED** [review]
